### ISSUE TITLE
"nc" regression in the new Agnhost containers are breaking the should provide Internet connection for containers tests?  

### ISSUE THREAD

**Issue by jayunit100** [kubernetes/kubernetes#98123]
tl;dr agnhost images on 2:26, dont work w/ ncat  anymore the same way..... 

```
➜  kubernetes git:(9ac1df0ebe4) ✗ kubectl logs    connectivity-test  -n nettest-3770                                           
Connection to 8.8.8.8 53 port [tcp/domain] succeeded!                                                                          
nc: getaddrinfo: Unrecognized service  
```

**What happened** 

- old image:     image: k8s.gcr.io/e2e-test-images/agnhost:2.21 ... passes

Looks like we have a test regression tracking to the introduction of new `agnhost` containers?   Not 100% sure but 

`git checkout 97cf67b3290` followed by 

```
make WHAT=test/e2e/e2e.test  && KUBECONFIG=/Users/jayunit100/.kube/config .//_output/local/go/bin/e2e.test --ginkgo.focus="should provide Internet connection for containers [Feature:Networking-IPv4]" 
```
Results in the following failed test... 

```
[Fail] [sig-network] Networking [It] should provide Internet connection for containers [Feature:Networking-IPv4] 
/Users/jayunit100/SOURCE/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/networking.go:97
```

Because containers are failed.

meanwhile https://testgrid.k8s.io/sig-network-gce#network-policies,%20google-gce shows that all tests are failing since friday, which is when the new agnhost merged ... 

**What you expected to happen**:

tests keep passing 50% of the time :) 

**How to reproduce it (as minimally and precisely as possible)**:

run  `--ginkgo.focus=should provide Internet connection for containers` tests with `--ginkgo.skip=6`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): head, kind, calico

**Comment by jayunit100** [comment]
@aojea  ... looks like We found that doing the one line change of moving agnhost to 2.21, fixed it...  here are the failing logs

```
Jan 17 11:24:29.538: INFO: b started at 2020-12-23 15:59:43 -0500 EST (0+4 container statuses recorded)                        
Jan 17 11:24:29.538: INFO:      Container cont-80-tcp ready: true, restart count 10                            
Jan 17 11:24:29.538: INFO:      Container cont-80-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-81-tcp ready: true, restart count 10                                            
Jan 17 11:24:29.538: INFO:      Container cont-81-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO: a started at 2020-12-23 15:59:44 -0500 EST (0+4 container statuses recorded)                        
Jan 17 11:24:29.538: INFO:      Container cont-80-tcp ready: true, restart count 10                        
Jan 17 11:24:29.538: INFO:      Container cont-80-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-81-tcp ready: true, restart count 10                           
Jan 17 11:24:29.538: INFO:      Container cont-81-udp ready: true, restart count 10                                            
Jan 17 11:24:29.538: INFO: local-path-provisioner-78776bfc44-n8h5d started at 2020-12-22 12:28:48 -0500 EST (0+1 container statuses recorded)
Jan 17 11:24:29.538: INFO:      Container local-path-provisioner ready: true, restart count 120                                
Jan 17 11:24:29.538: INFO: kube-proxy-wxqmp started at 2020-12-22 12:28:17 -0500 EST (0+1 container statuses recorded)                                                                                                                                         
Jan 17 11:24:29.538: INFO:      Container kube-proxy ready: true, restart count 10                                                                                                                                                                             
Jan 17 11:24:29.538: INFO: calico-node-zw7wl started at 2020-12-22 12:28:23 -0500 EST (3+1 container statuses recorded)                                                                                                                                        
Jan 17 11:24:29.538: INFO:      Init container upgrade-ipam ready: true, restart count 3
Jan 17 11:24:29.538: INFO:      Init container install-cni ready: true, restart count 0
Jan 17 11:24:29.538: INFO:      Init container flexvol-driver ready: true, restart count 0
Jan 17 11:24:29.538: INFO:      Container calico-node ready: true, restart count 12
Jan 17 11:24:29.538: INFO: c started at 2020-12-23 15:59:03 -0500 EST (0+4 container statuses recorded)
Jan 17 11:24:29.538: INFO:      Container cont-80-tcp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-80-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-81-tcp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-81-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO: c started at 2020-12-23 15:59:04 -0500 EST (0+4 container statuses recorded)
Jan 17 11:24:29.538: INFO:      Container cont-80-tcp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-80-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-81-tcp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-81-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO: coredns-f9fd979d6-wwcz8 started at 2020-12-22 12:28:39 -0500 EST (0+1 container statuses recorded)
Jan 17 11:24:29.538: INFO:      Container coredns ready: true, restart count 10
Jan 17 11:24:29.538: INFO: a started at 2020-12-23 15:59:03 -0500 EST (0+4 container statuses recorded)
Jan 17 11:24:29.538: INFO:      Container cont-80-tcp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-80-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-81-tcp ready: true, restart count 10
Jan 17 11:24:29.538: INFO:      Container cont-81-udp ready: true, restart count 10
Jan 17 11:24:29.538: INFO: connectivity-test started at 2021-01-17 11:23:58 -0500 EST (0+1 container statuses recorded)
Jan 17 11:24:29.538: INFO:      Container agnhost-container ready: false, restart count 0
W0117 11:24:29.544989   49552 metrics_grabber.go:105] Did not receive an external client interface. Grabbing metrics from ClusterAutoscaler is disabled.
Jan 17 11:24:29.791: INFO: 
Latency metrics for node calico-worker2
Jan 17 11:24:29.791: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "nettest-9707" for this suite.

• Failure [31.179 seconds]
[sig-network] Networking
/Users/jayunit100/SOURCE/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/framework.go:23
  should provide Internet connection for containers [Feature:Networking-IPv4] [It]
  /Users/jayunit100/SOURCE/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/networking.go:98

  Jan 17 11:24:28.958: Unexpected error:
      <*errors.errorString | 0xc0026ca2c0>: {
          s: "pod \"connectivity-test\" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-01-17 11:23:58 -0500 EST Reason: Message:} {Type:Ready Status:False Las
tProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-01-17 11:23:58 -0500 EST Reason:ContainersNotReady Message:containers with unready status: [agnhost-container]} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC
 LastTransitionTime:2021-01-17 11:23:58 -0500 EST Reason:ContainersNotReady Message:containers with unready status: [agnhost-container]} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-01-17 11:23:58 -050
0 EST Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:172.18.0.3 PodIP:192.168.88.45 PodIPs:[{IP:192.168.88.45}] StartTime:2021-01-17 11:23:58 -0500 EST InitContainerStatuses:[] ContainerStatuses:[{Name:agnhost-container State:{Waiting:nil R
unning:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2021-01-17 11:23:59 -0500 EST,FinishedAt:2021-01-17 11:23:59 -0500 EST,ContainerID:containerd://740fefa5d8ac3da14870ce83036653ce8bd648e92d1889c2602816c6bd2
d2581,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:k8s.gcr.io/e2e-test-images/agnhost:2.26 ImageID:k8s.gcr.io/e2e-test-images/agnhost@sha256:a3f7549ea04c419276e8b84e90a515bbce5bc8a057be2ed974ec45492eca3
46e ContainerID:containerd://740fefa5d8ac3da14870ce83036653ce8bd648e92d1889c2602816c6bd2d2581 Started:0xc0026c01cf}] QOSClass:BestEffort EphemeralContainerStatuses:[]}",
      }
      pod "connectivity-test" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-01-17 11:23:58 -0500 EST Reason: Message:} {Type:Ready Status:False LastProbeTime
:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-01-17 11:23:58 -0500 EST Reason:ContainersNotReady Message:containers with unready status: [agnhost-container]} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTrans
itionTime:2021-01-17 11:23:58 -0500 EST Reason:ContainersNotReady Message:containers with unready status: [agnhost-container]} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2021-01-17 11:23:58 -0500 EST Reas
on: Message:}] Message: Reason: NominatedNodeName: HostIP:172.18.0.3 PodIP:192.168.88.45 PodIPs:[{IP:192.168.88.45}] StartTime:2021-01-17 11:23:58 -0500 EST InitContainerStatuses:[] ContainerStatuses:[{Name:agnhost-container State:{Waiting:nil Running:nil
 Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2021-01-17 11:23:59 -0500 EST,FinishedAt:2021-01-17 11:23:59 -0500 EST,ContainerID:containerd://740fefa5d8ac3da14870ce83036653ce8bd648e92d1889c2602816c6bd2d2581,}} L
astTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:k8s.gcr.io/e2e-test-images/agnhost:2.26 ImageID:k8s.gcr.io/e2e-test-images/agnhost@sha256:a3f7549ea04c419276e8b84e90a515bbce5bc8a057be2ed974ec45492eca346e Contai
nerID:containerd://740fefa5d8ac3da14870ce83036653ce8bd648e92d1889c2602816c6bd2d2581 Started:0xc0026c01cf}] QOSClass:BestEffort EphemeralContainerStatuses:[]}
  occurred



```

**Comment by jayunit100** [comment]
```
➜  kubernetes git:(9ac1df0ebe4) ✗ kubectl logs    connectivity-test  -n nettest-3770
Connection to 8.8.8.8 53 port [tcp/domain] succeeded!
nc: getaddrinfo: Unrecognized service
```

**Comment by jayunit100** [comment]
@aojea this is the culprit :) 

https://github.com/kubernetes/kubernetes/pull/96918/files#diff-82e9acf13d70163bf94a3e122bd91591681b1ca14d072c4f90649cfb38a8242cR1 

It looks like whats going on is 
```
func checkConnectivityToHost(f *framework.Framework, nodeName, podName, host string, port, timeout int) error {
        command := []string{
                "nc",
                "-vz",
                "-w", strconv.Itoa(timeout),
                host,
                strconv.Itoa(port),
        }
```

Is failing, and the reason  , is that  the **NEW** alpine i guess doesnt have the same behaviour for `getaddrinfo`.  We just reprod'd it on zoom, reverted the agnhost change by just modifying the one line in `test/utils/image/manifest.go`,

```
-       configs[Agnhost] = Config{promoterE2eRegistry, "agnhost", "2.21"}
+       configs[Agnhost] = Config{promoterE2eRegistry, "agnhost", "2.26"}
```

and repro'd the failure.  

thanks @knabben for helping me figure this out !

**Comment by jayunit100** [comment]
oops this is a total dupe of #98117
